[cmd/builder] ocb fails on some environments due to go.mod issue

**Describe the bug**
ocb fails to build collector-contrib in Github workspace due to failing to download go 1.22 (not a specific distro/version)
<!-- A clear and concise description of what the bug is. -->

**Steps to reproduce**
<!-- If possible, provide a recipe for reproducing the error. -->
open collector-contrib on github workspace and run make e2e test
```
opentelemetry-collector-contrib git:(main) make e2e-test

/home/bits/go/src/github.com/DataDog/opentelemetry-collector-contrib/.tools/builder --skip-compilation --config cmd/otelcontribcol/builder-config.yaml --output-path cmd/otelcontribcol
Flag --output-path has been deprecated, use config distribution::output_path
2024-12-10T21:14:11.753Z	INFO	internal/command.go:97	OpenTelemetry Collector Builder	{"version": "v0.115.1-0.20241206185113-3f3e208e71b8"}
2024-12-10T21:14:11.753Z	INFO	internal/command.go:102	Using config file	{"path": "cmd/otelcontribcol/builder-config.yaml"}
2024-12-10T21:14:11.758Z	INFO	builder/config.go:162	Using go	{"go-executable": "/usr/local/go/bin/go"}
2024-12-10T21:14:11.766Z	INFO	builder/main.go:99	Sources created	{"path": "cmd/otelcontribcol"}
Error: failed to update go.mod: go subcommand failed with args '[mod tidy -compat=1.22]': exit status 1, error message: go: downloading go1.22 (linux/amd64)
go: download go1.22 for linux/amd64: toolchain not available

make: *** [Makefile:347: genotelcontribcol] Error 1
```
**What did you expect to see?**
e2e test runs sucessfully
<!-- A clear and concise description of what you expected to see. -->

**What did you see instead?**
see error message above
<!-- A clear and concise description of what you saw instead. -->

**What version did you use?**
<!-- Version: (e.g., `v0.4.0`, `1eb551b`, etc) -->

**What config did you use?**
<!-- Config: (e.g. the yaml config file) -->

**Environment**
Github workspace
<!-- OS: (e.g., "Ubuntu 20.04") -->
<!-- Compiler(if manually compiled): (e.g., "go 14.2") -->

**Additional context**
as @ankitpatel96 pointed out to me, it seems that from Go 1.21 and onward, there needs to be a sub-version specified in go.mod files. https://github.com/golang/go/issues/62278
Also, all the other go.mod files in collector repo have a 1.22.0 specification, so this seems like an easy, uncontroversial fix.
<!-- Add any other context about the problem here. -->

--- a/cmd/builder/README.md
+++ b/cmd/builder/README.md
@@ -96,6 +96,8 @@ go install go.opentelemetry.io/collector/cmd/builder@latest
 
 If installing through this method the binary will be called `builder`.
 
+Please note that the `go.mod` file generated uses `go 1.22` as the version. Versions 1.22.3, 1.21.10, and prior of Go [do not recognize this as a valid go version](https://github.com/golang/go/commit/27ed85d4d1702e868730ab6ea2ad6326988c615c). In order to successfully generate and build a collector using ocb, you must use Go version 1.22.4+, or any version of Go 1.23 and beyond.
+
 ## Running
 
 A build configuration file must be provided with the `--config` flag.

